app/src/main/java/com/sunasterisk/englishflashcard/ui/addenglish/AddEnglishFragment.kt

```diff
@@ -1,13 +1,90 @@
 package com.sunasterisk.englishflashcard.ui.addenglish
 
+import android.annotation.SuppressLint
+import android.app.AlertDialog
 import com.sunasterisk.englishflashcard.R
+import com.sunasterisk.englishflashcard.data.model.Topic
+import com.sunasterisk.englishflashcard.data.repository.TopicsRepository
+import com.sunasterisk.englishflashcard.data.source.local.TopicsLocalDataSource
+import com.sunasterisk.englishflashcard.data.source.local.dao.TopicsDAOlmpl
 import com.sunasterisk.englishflashcard.ui.BaseFragment
+import com.sunasterisk.englishflashcard.ui.learnandsearch.TopicAdapter
+import com.sunasterisk.englishflashcard.ui.learnandsearch.TopicContract
+import com.sunasterisk.englishflashcard.ui.learnandsearch.TopicPresenter
+import com.sunasterisk.englishflashcard.ui.toast
+import kotlinx.android.synthetic.main.dialog_custom.view.*
+import kotlinx.android.synthetic.main.fragment_add_english.*
+import java.lang.Exception
 
-class AddEnglishFragment : BaseFragment() {
+class AddEnglishFragment : BaseFragment(), TopicContract.View, AddEnglishContract.View {
+
+    private var topicAdapter = TopicAdapter()
+    private var topicPresenter: TopicContract.Presenter? = null
+    private var addEnglishPresenter: AddEnglishContract.Presenter? = null
 
     override val layoutResource get() = R.layout.fragment_add_english
 
     override fun initComponents() {
+        initDataTopics()
+        addTopic()
+    }
+
+    private fun initDataTopics() {
+        context?.let {
+            val topicsDao = TopicsDAOlmpl.getInstance(it)
+            val localRepository: TopicsRepository = TopicsRepository.getInstance(
+                local = TopicsLocalDataSource.getInstance(topicsDao)
+            )
+            topicPresenter = TopicPresenter(this, localRepository)
+            topicPresenter?.getAllTopics()
+        }
+        recyclerViewTopics.adapter = topicAdapter
+    }
+
+    @SuppressLint("InflateParams")
+    fun addTopic() {
+        buttonAddTopic.setOnClickListener {
+            val builder = AlertDialog.Builder(context, R.style.CustomAlertDialog)
+            val view = layoutInflater.inflate(R.layout.dialog_custom, null)
+            builder.setView(view)
+            val alertDialog: AlertDialog = builder.create()
+            alertDialog.show()
```

Comment:
viết tách ra hàm mới, 
showABCDialog
Đặt tên lại cho phù hợp 
```suggestion
            AlertDialog.Builder(context, R.style.CustomAlertDialog)
                    .setView(layoutInflater.inflate(R.layout.dialog_gì_đây, null))
                    .create()
                    .show()
```